File: src/main/java/uk/co/alt236/apkcompare/comparators/results/ResultBlock.java
package uk.co.alt236.apkcompare.comparators.results;

import java.util.List;

public class ResultBlock implements ComparisonResult {

    private final List<ResultItem> resultItems;
    private final String title;

    public ResultBlock(String title,
                       List<ResultItem> resultItems) {
        this.title = title;
        this.resultItems = resultItems;
    }

    public List<ResultItem> getResultItems() {
        return resultItems;
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public Similarity getSimilarity() {
        return CollectionSimilarityEvaluator.evaluate(resultItems);
    }

}

File: app/src/main/java/uk/co/alt236/apkcompare/app/comparators/results/ResultBlock.kt
package uk.co.alt236.apkcompare.app.comparators.results

data class ResultBlock @JvmOverloads constructor(override val title: String,
                                                 val comparisonResults: List<ComparisonResult>,
                                                 val complex: Boolean = false) : ComparisonResult {

    override val similarity: Similarity by lazy {
        CollectionSimilarityEvaluator.evaluate(comparisonResults)
    }
}
